src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -225,14 +245,45 @@ protected function writeToCsv(\PDOStatement $stmt, CsvFile $csv)
 
             // write the rest
             $numRows = 1;
+            $lastRow = null;
             while ($resultRow = $stmt->fetch(\PDO::FETCH_ASSOC)) {
                 $csv->writeRow($resultRow);
+                $lastRow = $resultRow;
                 $numRows++;
             }
-
-            return $numRows;
+            if ($this->incrementalFetching) {
+                if (isset($this->incrementalFetching['autoIncrementColumn'])) {
+                    if (!array_key_exists($this->incrementalFetching['autoIncrementColumn'], $lastRow)) {
+                        throw new UserException(
+                            sprintf(
+                                "Specified autoIncrement column %s not found in table",
+                                $this->incrementalFetching['autoIncrementColumn']
+                            )
+                        );
+                    }
+                    $output['lastFetchedRow']['autoIncrement'] = $lastRow[$this->incrementalFetching['autoIncrementColumn']];
```

Comment:
This will require ordering in the query, and also, won't provide the last id if the ordering is by timestamp.
Either get the value from the table metadata, or change this to use *either* autoIncrement column *or* timestamp, and order by the column used.